k8s/upgrades/0.5.x-0.6.0/oebs_update.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+################################################################
+# STEP: Get Persistent Volume (PV) name as argument            #
+#                                                              #
+# NOTES: Obtain the pv to upgrade via "kubectl get pv"         #
+################################################################
+
+if [ "$#" -ne 2 ]; then
+    echo 
+    echo "Usage:"
+    echo 
+    echo "$0 <pv-name> <node-label>"
+    echo 
+    echo "  <pv-name> Get the PV name using: kubectl get pv"
+    echo "  <node-label> Label applied to the nodes where replicas of"
+    echo "    this PV are present. Get the nodes by running:"
+    echo "    kubectl get pods --all-namespaces -o wide | grep <pv-name>"
+    exit 1
+fi
+
+pv=$1
+replica_node_label=$2
+
+pvc=`kubectl get pv $pv -o jsonpath="{.spec.claimRef.name}"`
+ns=`kubectl get pv $pv -o jsonpath="{.spec.claimRef.namespace}"`
+
+################################################################ 
+# STEP: Generate deploy, replicaset and container names from PV#
+#                                                              #
+# NOTES: Ex: If PV="pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc", #
+#                                                              #
+# ctrl-dep: pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc-ctrl      #  
+# ctrl-cont: pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc-ctrl-con #  
+################################################################
+
+c_dep=$(echo $pv-ctrl); c_name=$(echo $c_dep-con)
+r_dep=$(echo $pv-rep); r_name=$(echo $r_dep-con)
+rep_count=`kubectl get deploy $r_dep --namespace $ns -o jsonpath="{.spec.replicas}"`
+
+echo "Patching Replica Deployment upgrade strategy as recreate"
+kubectl patch deployment --namespace $ns --type json $r_dep -p "$(cat patch-strategy-recreate.json)"
+rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+
+echo "Patching Controller Deployment upgrade strategy as recreate"
+kubectl patch deployment --namespace $ns --type json $c_dep -p "$(cat patch-strategy-recreate.json)"
+rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+
+c_rs=$(kubectl get rs -o name --namespace $ns | grep $c_dep | cut -d '/' -f 2)
```

Comment:
can we write some comment on what is being done.
can we make use of jsonpath etc to get this done easily e.g. making use of `labelselector` option

Reply:
done! thank you.